lib/qesapdeployment.pm

```diff
@@ -601,6 +601,17 @@ sub qesap_execute_conditional_retry {
         if ($ret[0]) {
             if (qesap_file_find_string(file => $ret[1], search_string => $args{error_string})) {
                 record_info('DETECTED ' . uc($args{cmd}) . ' ERROR', $args{error_string});
+
+                # Executing terraform destroy before retrying terraform apply
+                if ($args{destroy_terraform}) {
```

Comment:
new argument has to be documented in https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/20786/files#diff-23c988d39d71dc252bb5dce95a699047e93941078f2152ab4a6587286e9a4e47R583

Reply:
Please add a new unit test in t/09_qesapdeployment.t calling the function with `destroy_terraform => 1`